src/backend/dx11/src/device.rs

```diff
@@ -904,10 +904,15 @@ impl hal::Device<Backend> for Device {
         })
     }
 
-    fn create_pipeline_cache(&self) -> Result<(), device::OutOfMemory> {
+    fn create_pipeline_cache(&self, _data: Option<Vec<u8>>, device: &PhysicalDevice) -> Result<(), device::OutOfMemory> {
```

Comment:
why do we need the physical device here? It's not in https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/vkCreatePipelineCache.html

Reply:
I couldn't find other way to access the `PhysicalDeviceProperties`  (for the cache validation) from the Device, so I placed the `&PhysicalDevice` here.
I also meant to propagate the validation to the create users, but they can't access backend specific resources e.g. `vk::PipelineCacheHeaderVersion` or `properties.pipeline_cache_uuid` from `gfx-hal`.
My other idea is to move the cache validation into the `PhysicalDevice` and the crate users can call it (e.g. `adpater.physical_device.is_valid_cache(&cache_bytes)`) before `create_pipeline_cache`.

Reply:
We don't have to, strictly speaking, validate the caches on `create_pipeline_cache`, at least not at the API level. The backend can still keep VkPhysicalDevice around when creating a logical device and use it for invalidation.